snorkel/labeling/model/label_model.py

```diff
@@ -368,10 +369,124 @@ def predict_proba(self, L: sparse.spmatrix) -> np.ndarray:
         Z = np.tile(X.sum(axis=1).reshape(-1, 1), self.cardinality)
         return X / Z
 
+    def _break_ties(
+        self, Y_prob: np.ndarray, break_ties: Optional[str] = "abstain"
+    ) -> np.ndarray:
+        """Break ties among probabilistic labels according to given policy.
+
+        Policies to break ties include:
+        "abstain": return an abstain vote (0)
+        "random": randomly choose among the tied options
+        NOTE: if break_ties="random", repeated runs may have slightly different results due to difference in broken ties
+
+        Parameters
+        ----------
+        Y_prob
+            An [n,k] array of probabilistic labels
+        break_ties
+            Policy to break ties, by default 'random'
+
+        Returns
+        -------
+        np.ndarray
+            An [n,k] array of integer labels
+        """
+
+        n, k = Y_prob.shape
+        Y_pred = np.zeros(n)
+        diffs = np.abs(Y_prob - Y_prob.max(axis=1).reshape(-1, 1))
+
+        TOL = 1e-5
+        for i in range(n):
+            max_idxs = np.where(diffs[i, :] < TOL)[0]
+            if len(max_idxs) == 1:
+                Y_pred[i] = max_idxs[0] + 1
+            # Deal with "tie votes" according to the specified policy
+            elif break_ties == "random":
+                Y_pred[i] = np.random.choice(max_idxs) + 1
+            elif break_ties == "abstain":
+                Y_pred[i] = 0
+            else:
+                raise ValueError(f"break_ties={break_ties} policy not recognized.")
+        return Y_pred
+
+    def predict(
+        self, L: sparse.spmatrix, break_ties: Optional[str] = "abstain"
```

Comment:
I'm torn here. If the default is abstain, then there's a potential sneaky bug where a person gets way more wrong than they should (since the gold will almost never be abstain). But if the default is random, then a person not knowing that we have random tie-breaking could see that their score is stochastic and freak out. MajorityLabelVoter() just uses vanilla `argmax`, which will take the first option if there's a tie. Could we do the same here and make that the default? If multiple classes are tied, it returns the first class in the tie?

Reply:
We could pass in an option for a label to default to in case of a tie, say always the negative class for example. Returning the first class probably makes sense to match MV, but it feels like introducing weird biases?

Reply:
Created `random` and `true-random` options for tie breaking policy. true-random is non-deterministic and `random` is deterministic random